experimental/uefi/baremetal/runner

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Thin shell script invoked as a cargo runner to run the compiled efi firmware
+# in QEMU. Detects if kvm is supported, and sets qemu flags based on that.
+# Instead of this single runner script it would be preferable to use a different
+# runner based on whether the kvm feature is set. However, cargo does not
+# currently allow this. Ref: https://github.com/rust-lang/cargo/issues/8170
+
+readonly TARGET=$1
+
+qemu_flags=(
+  '-nodefaults'
+  '-nographic'
+  '-no-reboot'
+  '-serial' 'file:/dev/stdout'
+  '-serial' 'stdio'
+  '-machine' 'q35'
+  '-device' 'isa-debug-exit,iobase=0xf4,iosize=0x04'
+)
+
+# Use kvm if supported, as it is required for certain features. Note
+# that hosts that support kvm still need to disable the default
+# `support_emulated_runner` feature in cargo to include code that requires kvm
+# in the compiled firmware. Ideally this check would rely on the cargo
+# flag itself, enabling kvm if it is disabled. However cargo does not expose
+# which flags are set to the runner. Ref: https://doc.rust-lang.org/cargo/reference/environment-variables.html
+if [[ -e "/dev/kvm" ]]; then
+  qemu_flags+=(
+    '-enable-kvm'
+    '-cpu' 'IvyBridge-IBRS,enforce'
+  )
+fi
```

Comment:
I'm not sure we need this logic to detect KVM support for the bare metal binary. Should instead always pass the kvm flags? 

We use this script in the UEFI app only because we wanted to keep running existing tests in the GitHub CI (which does not support KVM). However for this app it may be fine to mandate KVM support in the host? Would simply this a bit.

Reply:
Unfortunately for now we do. Remember that all tests are executed with `qemu` as well, therefore if we require KVM here, `cargo test` will fail unless KVM is available -- and this also means CI will fail under github actions.

Once we ensure that we have access to KVM in CI (and dev container), we can indeed drop that.

Reply:
Fair, if qemu in emulation mode works for running the tests it's probs a good idea to keep this around. :) Might be worth removing the references to the `support_emulated_runner` feature and efi firmware firmware from the comments though, since they're specific to the UEFI app.